src/csharp/Microsoft.Spark/ML/Linalg/SparseVectorUDT.cs

```diff
@@ -0,0 +1,75 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Spark.Sql;
+using Microsoft.Spark.Sql.Types;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Spark.ML.Linalg
+{
+    /// <summary>
+    /// The data type for a SparseVectorUDT, used by SparseVector. 
+    /// </summary>
+    public class SparseVectorUDT : DataType
+    {
+        /// <summary>
+        /// Returns JSON object describing this type.
+        /// </summary>
+        internal override object JsonValue =>
+            new JObject(
+                new JProperty("type", "udt"),
+                new JProperty("class", "org.apache.spark.ml.linalg.VectorUDT"),
+                new JProperty("pyClass", "pyspark.ml.linalg.VectorUDT"),
+                new JProperty("sqlType", new JObject(
+                    new JProperty("type", "struct"),
+                    new JProperty("fields", new JArray(
+                            new JObject(
+                                new JProperty("name", "type"),
+                                new JProperty("type", "byte"),
+                                new JProperty("nullable", false)
+                                ),
+                            new JObject(
+                                new JProperty("name", "size"),
+                                new JProperty("type", "integer"),
+                                new JProperty("nullable", false)
+                            ),
+                            new JObject(
+                                new JProperty("name", "indices"),
+                                new JProperty("type", new JObject(
+                                    new JProperty("type", "array"),
+                                    new JProperty("elementType", "integer"),
+                                    new JProperty("containsNull", false))),
+                                new JProperty("nullable", true)
+                            ),
+                            new JObject(
+                                new JProperty("name", "values"),
+                                new JProperty("type", new JObject(
+                                    new JProperty("type", "array"),
+                                    new JProperty("elementType", "double"),
+                                    new JProperty("containsNull", false))),
+                                new JProperty("nullable", true)
+                            )
+                        ))
+                    ))
+                );
+
+        internal override object FromInternal(object obj)
+        {
+            if (obj is RowConstructor rowConstructor)
+            {
+                return rowConstructor.GetRow();
+            }
+
+            return obj;
+        }
+
+        internal override bool NeedConversion() => true;
```

Comment:
I don't see where the `SparseVectorUDT` class is being used, but why do you need this ?  Does this get unpickled as a `Row` ?

Reply:
The class is for users so they don't have to define the schema anytime they want to use it - the alternative is that they would have to know to send a vector from their app to spark would have to know the type of vector (sparse or dense) and the schema they need to send it across.

Reply:
Can I see an example of how this is going to be used ?